src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -235,20 +235,38 @@ void MeshForwarder::UpdateIndirectMessages(void)
     }
 }
 
-otError MeshForwarder::EvictIndirectMessage(void)
+otError MeshForwarder::EvictMessage(uint8_t aPriority)
 {
-    otError error = OT_ERROR_NOT_FOUND;
+    otError  error = OT_ERROR_NOT_FOUND;
+    Message *message;
 
-    for (Message *message = mSendQueue.GetHead(); message; message = message->GetNext())
-    {
-        if (!message->IsChildPending())
-        {
-            continue;
-        }
+    VerifyOrExit((message = mSendQueue.GetTail()) != NULL);
 
+    if (message->GetPriority() > aPriority)
+    {
         RemoveMessage(*message);
         ExitNow(error = OT_ERROR_NONE);
     }
+    else if (message->GetPriority() == aPriority)
```

Comment:
The condition is too strict, there might be no chance to evict the higher priority indirect message.

Reply:
You are right. This condition should be removed.